time-series/time-series-api/src/test/java/com/powsybl/timeseries/TimeSeriesTest.java

```diff
@@ -498,4 +515,29 @@ void splitTest() {
         assertArrayEquals(new double[] {Double.NaN, Double.NaN, 3d}, split.get(1).get(0).toArray(), 0d);
         assertArrayEquals(new double[] {Double.NaN, Double.NaN, 6d}, split.get(1).get(1).toArray(), 0d);
     }
+
+    private static Stream<Arguments> getArgumentsWriteInstantToString() {
+        return Stream.of(
+            Arguments.of(Instant.ofEpochSecond(123456, 7), "123456000000007", 9),
+            Arguments.of(Instant.ofEpochSecond(123456, 7000), "123456000007000", 9),
+            Arguments.of(Instant.ofEpochSecond(123456, 700000000), "123456700000000", 9),
+            Arguments.of(Instant.ofEpochSecond(0, 7), "7", 9),
+            Arguments.of(Instant.ofEpochSecond(0, 7000), "7000", 9),
+            Arguments.of(Instant.ofEpochSecond(0, 700000000), "700000000", 9),
+            Arguments.of(Instant.ofEpochSecond(123456, 7), "123456000000", 6),
+            Arguments.of(Instant.ofEpochSecond(123456, 700), "123456000001", 6), // Case with rounding
+            Arguments.of(Instant.ofEpochSecond(123456, 7000), "123456000007", 6),
+            Arguments.of(Instant.ofEpochSecond(123456, 700000000), "123456700000", 6),
+            Arguments.of(Instant.ofEpochSecond(0, 7), "0", 6),
+            Arguments.of(Instant.ofEpochSecond(0, 700), "1", 6), // Case with rounding
+            Arguments.of(Instant.ofEpochSecond(0, 7000), "7", 6),
+            Arguments.of(Instant.ofEpochSecond(0, 700000000), "700000", 6)
```

Comment:
Related to my previous comment:
```suggestion
            Arguments.of(Instant.ofEpochSecond(0, 700000000), "700000", 6),
            Arguments.of(Instant.ofEpochSecond(0, 999999999), "1000000", 6)
```